cpp/include/cudf/null_mask.hpp

```diff
@@ -113,13 +113,13 @@ void set_null_mask(bitmask_type* bitmask,
                    rmm::cuda_stream_view stream = cudf::get_default_stream());
 
 /**
- * @brief Sets a vector of pre-allocated bitmask buffers to given states in the corresponding ranges
- * in bulk
+ * @brief Sets a vector of non-overlapping pre-allocated bitmask buffers to given states in the
+ * corresponding ranges in bulk
  *
  * Sets bit ranges `[begin_bit, end_bit)` of given bitmasks to specified valid states. The bitmask
- * bit ranges must be non-aliasing. i.e., attempting to concurrently set bits within the same
- * physical word across bitmasks will result in undefined behavior. This utility is optimized for
- * bulk operation on 16 or more bitmasks sized 2^24 bits or less.
+ * bit ranges must be non-overlapping. i.e., attempting to set a physical bit concurrently across
```

Comment:
Aliasing now handled by this API with a small performance hit. The original functionality moved to the `_unsafe` API.

Reply:
It does seem like you are changing the behavior of the API and callers may be expecting the faster (unsafe) implementation so this change would be a surprise. You could deprecate the API and have it call the unsafe function which would match the current behavior. And introduce the new function as `cudf::set_null_masks_safe`

Reply:
To align with our bit utilities e.g. https://github.com/rapidsai/cudf/blob/d953676e9281125a5b8bd9be739c997611471771/cpp/include/cudf/utilities/bit.hpp
I believe it's more appropriate to make this a breaking change: "safe" behavior should be the default, while the unsafe variant should be explicitly marked with an `_unsafe` suffix, rather than the other way around.

Reply:
> It does seem like you are changing the behavior of the API and callers may be expecting the faster (unsafe) implementation so this change would be a surprise.

One more thing here is that this API is fairly recent (added in 25.06) and I don't think it's being used anywhere in libcudf yet either and I doubt it's being used outside of libcudf either but we can't make such assumptions.

Reply:
> If we make the default as safe then we don't even need to have an API with safe suffix altogether as the set_null_masks and set_null_masks_unsafe pair makes sense as is

That’s how I’d prefer it too.

Reply:
The problem is that you are changing the behavior of the current API. It behaved as documented -- it did not contain a bug that we are fixing but rather worked as advertised. Now you want a newer safe version of this API. It seems more correct to add the new API (_safe) and deprecate the old one without changing the old one's behavior.

Reply:
I have updated the PR as per @davidwendt's recommendation. The original API is deprecated and points to the `_unsafe` version internally while a `_safe` API has been added.